src/bridge/IrcBridge.ts

```diff
@@ -165,6 +167,17 @@ export class IrcBridge {
             this.initialiseMetrics();
         }
         this.publicitySyncer = new PublicitySyncer(this);
+
+
+        const homeserverToken = this.registration.getHomeserverToken();
+        if (!homeserverToken) {
+            throw Error("No HS token defined");
+        }
+
+        this.appservice = new AppService({
+            homeserverToken,
+            httpMaxSizeBytes: (this.config.advanced || { }).maxTxnSize || TXN_SIZE_DEFAULT,
```

Comment:
I would rather ensure that `this.config.advanced` is always at least `{}` rather than this

Reply:
Yep, we generally need a default config to rely on. This will do for now.